ionoscloud/data_source_dbaas_pgsql_cluster.go

```diff
@@ -194,7 +206,13 @@ func dataSourceDbaasPgSqlReadCluster(ctx context.Context, d *schema.ResourceData
 			return diags
 		}
 	} else {
-		var results []dbaas.ClusterResponse
+		//var results []dbaas.ClusterResponse
+		clusters, _, err := client.ListClusters(ctx, "")
```

Comment:
if I will move the line: ```clusters, _, err := client.ListClusters(ctx, name)``` in the else of partial match, if you not provide a name(so if you search only after location for example), the results slice will be empty since the results variable is populated only in the if name if part, and you can not find any data source based on other parameters.

Reply:
Yes, this is why i proposed 

> On the else branch of nameOk you can make another ListClusters call to be sure that result have all clusters for the filters after.